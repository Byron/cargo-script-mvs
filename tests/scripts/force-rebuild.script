#!/bin/sh
set -e -u

# https://unix.stackexchange.com/questions/30091/fix-or-alternative-for-mktemp-in-os-x
mytmpdir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')

cd "$mytmpdir"

printf 'let msg = option_env!("_RUST_SCRIPT_TEST_MESSAGE").unwrap_or("undefined"); println!("msg = {}", msg);' > script.rs

rust-script script.rs

export _RUST_SCRIPT_TEST_MESSAGE=hello

rust-script script.rs

rust-script --force script.rs
