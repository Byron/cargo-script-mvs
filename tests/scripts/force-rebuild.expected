msg = undefined
msg = undefined
msg = hello
